moved/src/move_execution/mod.rs

```diff
@@ -116,33 +116,76 @@ where
     vm.new_session_with_extensions(state, native_extensions)
 }
 
-#[allow(clippy::too_many_arguments)]
-pub fn execute_transaction(
-    tx: &NormalizedExtendedTxEnvelope,
-    tx_hash: &B256,
-    state: &(impl MoveResolver<PartialVMError> + TableResolver),
-    genesis_config: &GenesisConfig,
-    l1_cost: u64,
-    l2_fee: impl L2GasFee,
-    l2_input: L2GasFeeInput,
-    base_token: &impl BaseTokenAccounts,
-    block_header: HeaderForExecution,
+pub enum TransactionExecutionInput<
+    'input,
+    S: MoveResolver<PartialVMError> + TableResolver,
```

Comment:
Typically you don't put the trait bounds on the data type definition only on the function/method implementations. The reason is because trait bounds restrict _behaviour_, which is the domain of methods/functions, where as data type definitions are about _structure_ only. Keeping trait bounds only on behaviour definitions allows the opportunity to reuse data types where the same structure makes sense in multiple different behaviour settings.